File: Framework/CSharp/Framework/Framework/Net/SmartPop3ClientConfig.cs
/*
 * 项目：Smartkernel.Framework
 * 作者：曹艳白
 * 邮箱：caoyanbai@139.com
 */
using System;

namespace Smartkernel.Framework.Net
{
    /// <summary>
    /// 邮件Pop3客户端的配置实体
    /// </summary>
    [Serializable]
    public class SmartPop3ClientConfig
    {
        /// <summary>
        /// 私有构造函数
        /// </summary>
        public SmartPop3ClientConfig() {}

        /// <summary>
        /// 构造函数
        /// </summary>
        /// <param name="host">Pop3服务器地址</param>
        /// <param name="username">接收者用户名</param>
        /// <param name="password">接收者密码</param>
        /// <param name="port">Pop3服务器端口</param>
        public SmartPop3ClientConfig(string host, string username, string password, int port = 110)
        {
            Host = host;
            Port = port;
            Username = username;
            Password = password;
        }

        /// <summary>
        /// 获取或设置Pop3服务器地址
        /// </summary>
        public string Host { get; set; }

        /// <summary>
        /// 获取或设置Pop3服务器端口
        /// </summary>
        public int Port { get; set; }

        /// <summary>
        /// 获取或设置接收者用户名
        /// </summary>
        public string Username { get; set; }

        /// <summary>
        /// 获取或设置接收者密码
        /// </summary>
        public string Password { get; set; }
    }
}

File: Framework/Java/framework/framework/smartkernel/framework/net/SmartPop3ClientConfig.java
/*
 * 项目：Smartkernel.Framework
 * 作者：曹艳白
 * 邮箱：caoyanbai@139.com
 */
package smartkernel.framework.net;

import java.io.*;

/**
 * Pop3客户端配置实体
 *
 */
public class SmartPop3ClientConfig implements Serializable {
	private static final long serialVersionUID = 1L;

	/**
	 * 公有构造函数
	 */
	public SmartPop3ClientConfig() {
	}

	/**
	 * 构造函数
	 * 
	 * @param host
	 *            ，Pop3服务器地址
	 * @param username
	 *            ，登录Pop3服务器的用户名
	 * @param password
	 *            ，登录Pop3服务器的密码
	 * @param port
	 *            ，Pop3服务器端口
	 */
	public SmartPop3ClientConfig(String host, String username, String password,
			int port) {
		this.setHost(host);
		this.setPort(port);
		this.setUsername(username);
		this.setPassword(password);
	}

	/**
	 * 构造函数
	 * 
	 * @param host
	 *            ，Pop3服务器地址
	 * @param username
	 *            ，登录Pop3服务器的用户名
	 * @param password
	 *            ，登录Pop3服务器的密码
	 */
	public SmartPop3ClientConfig(String host, String username, String password) {
		this.setHost(host);
		this.setPort(110);
		this.setUsername(username);
		this.setPassword(password);
	}

	private String host;

	private int port;

	private String username;

	private String password;

	/**
	 * 获取Pop3服务器地址
	 * 
	 * @return，Pop3服务器地址
	 */
	public String getHost() {
		return host;
	}

	/**
	 * 设置Pop3服务器地址
	 * 
	 * @param host
	 *            ，Pop3服务器地址
	 */
	public void setHost(String host) {
		this.host = host;
	}

	/**
	 * 获取Pop3服务器端口
	 * 
	 * @return，Pop3服务器端口
	 */
	public int getPort() {
		return port;
	}

	/**
	 * 设置Pop3服务器端口
	 * 
	 * @param port
	 *            ，Pop3服务器端口
	 */
	public void setPort(int port) {
		this.port = port;
	}

	/**
	 * 获取登录Pop3服务器的用户名
	 * 
	 * @return，登录Pop3服务器的用户名
	 */
	public String getUsername() {
		return username;
	}

	/**
	 * 设置登录Pop3服务器的用户名
	 * 
	 * @param username
	 *            ，登录Pop3服务器的用户名
	 */
	public void setUsername(String username) {
		this.username = username;
	}

	/**
	 * 获取登录Pop3服务器的密码
	 * 
	 * @return，登录Pop3服务器的密码
	 */
	public String getPassword() {
		return password;
	}

	/**
	 * 设置登录Pop3服务器的密码
	 * 
	 * @param password
	 *            ，登录Pop3服务器的密码
	 */
	public void setPassword(String password) {
		this.password = password;
	}
}
